Assertion failed: _total_size == 0 && _count == 0 at line 30 of panda/src/gobj/bufferContextChain.I

## Description

Closing a `p3vulkandisplay` window on macOS by clicking its close button (or cmd+q) throws an assert.

**Note**: This bug is specific to the `vulkan` branch.

``` sh
➜  panda3d-playground PYTHONPATH=panda3d-vulkan-vanilla/built venv/bin/python3.12 minimal.py 
Known pipe types:
  VulkanGraphicsPipe
(all display modules loaded.)
Window open. Now close the runtime to trigger the assert.
Assertion failed: _total_size == 0 && _count == 0 at line 30 of panda/src/gobj/bufferContextChain.I
Assertion failed: (_next == nullptr && _prev == nullptr) || (_next == this && _prev == this) at line 60 of built/include/linkedListNode.I
Assertion failed: _total_size == 0 && _count == 0 at line 30 of panda/src/gobj/bufferContextChain.I
Assertion failed: (_next == nullptr && _prev == nullptr) || (_next == this && _prev == this) at line 60 of built/include/linkedListNode.I
Traceback (most recent call last):
  File "/Users/daniel/Dev/panda3d-playground/panda3d-vulkan-vanilla/built/direct/showbase/ShowBase.py", line 2264, in __igLoop
    self.graphicsEngine.renderFrame()
AssertionError: _total_size == 0 && _count == 0 at line 30 of panda/src/gobj/bufferContextChain.I
:task(error): Exception occurred in PythonTask igLoop
Traceback (most recent call last):
  File "/Users/daniel/Dev/panda3d-playground/minimal.py", line 16, in <module>
    app.run()
  File "/Users/daniel/Dev/panda3d-playground/panda3d-vulkan-vanilla/built/direct/showbase/ShowBase.py", line 3454, in run
    self.taskMgr.run()
  File "/Users/daniel/Dev/panda3d-playground/panda3d-vulkan-vanilla/built/direct/task/Task.py", line 629, in run
    self.step()
  File "/Users/daniel/Dev/panda3d-playground/panda3d-vulkan-vanilla/built/direct/task/Task.py", line 579, in step
    self.mgr.poll()
  File "/Users/daniel/Dev/panda3d-playground/panda3d-vulkan-vanilla/built/direct/showbase/ShowBase.py", line 2264, in __igLoop
    self.graphicsEngine.renderFrame()
AssertionError: _total_size == 0 && _count == 0 at line 30 of panda/src/gobj/bufferContextChain.I
```

## Steps to Reproduce

1. Switch to the `vulkan` branch and build
2. Run the script below `minimal.py`
3. Wait for the scene to render
4. Press cmd+q to close the runtime

**Result**: assertion seen above
**Expected Result**: process exits cleanly without any errors.

```python
from panda3d.core import load_prc_file_data
# p3vulkandisplay only exists on the vulkan branch; on master this module
# fails to load and Panda falls back to aux-display (pandagl) automatically.
load_prc_file_data('', 'load-display p3vulkandisplay\naux-display pandagl')

from direct.showbase.ShowBase import ShowBase

class MinimalApp(ShowBase):
    def __init__(self):
        ShowBase.__init__(self)
        print("Window open. Now close the runtime to trigger the assert.")

    def idleTask(self, task):
        return task.cont

app = MinimalApp()
app.taskMgr.add(app.idleTask, "IdleTask")
app.run()
```

## Environment
 * Operating system: macOS Tahoe 26.5.1 (build 25F80)
 * System architecture: arm64 (Apple Silicon)
 * Panda3D version: 1.11.0 (dev build)
 * Installation method: built from source, `vulkan` branch
 * Python version (if using Python): 3.12 (built/run against a Homebrew-framework Python via `venv`)
 * Compiler (if using C++): Xcode 26.6 (Build version 17F113)

--- a/panda/src/cocoadisplay/cocoaGraphicsWindow.mm
+++ b/panda/src/cocoadisplay/cocoaGraphicsWindow.mm
@@ -1578,36 +1578,12 @@ handle_close_event() {
 
   _window = nil;
 
-  // Get rid of the GSG
-  if (_gsg != nullptr) {
-    unbind_context();
-    _gsg.clear();
-  }
-
-  // Dump the view, too
-  if (_view != nil) {
-    [_view release];
-    _view = nil;
-  }
-
-  // Unhide the mouse cursor
-  if (_mouse_hidden) {
-    [NSCursor unhide];
-    _mouse_hidden = false;
-  }
-
-  // Release the cursor.
-  if (_cursor != nil) {
-    [_cursor release];
-    _cursor = nil;
-  }
+  close_window();
 
   WindowProperties properties;
   properties.set_open(false);
   properties.set_cursor_hidden(false);
   system_changed_properties(properties);
-
-  GraphicsWindow::close_window();
 }
 
 /**

